src/main/kotlin/com/hibob/kotlinBasic/sequences/Sequences.kt

```diff
@@ -0,0 +1,91 @@
+package com.hibob.kotlinBasic.sequences
+
+fun ex1() {
+    // change the program to
+    // 1. reuse the filter / map function
+    // 2. println each call to track the diffs between List and Seq
+
+    val list = listOf(1, 2, 3, 4)
+
+    println("List:")
+    val maxOddSquare = list
+        .map { powerOfTwo(it) }
+        .filter { isEven(it) }
+        .find { it == 4 }
+
+    println("-------------------------")
+
+    println("Sequence:")
+    val maxOddSquare2 = list.asSequence()
+        .map { powerOfTwo(it) }
+        .filter { isEven(it) }
+        .find { it == 4 }
+
+}
+
+private fun powerOfTwo(number: Int): Int {
+    println("powerOfTwo was called with number: $number")
+    return number * number
+}
+
+private fun isEven(number: Int): Boolean {
+    println("isEven was called with number: $number")
+    return number % 2 == 0
+}
+
+fun ex2() {
+    // how many times filterFunc was called
+    fun filterFunc(it: Int): Boolean {
+        println("filterFunc was called")
+        return it < 3
+    }
+    sequenceOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+        .filter { filterFunc(it) }
+}
+
+fun ex3() {
+    // create the list of the first 10 items of (2, 4, 8, 16, 32 ...) seq
+
+    val list = generateSequence(2) { it * 2 }
+        .take(10)
+        .toList()
+
+    println(list)
+}
+
+fun ex4() {
+    // create the list of the first 10 items of the Fibonacci seq
+
+    val fibonacciList = generateSequence(Pair(1, 1)) { Pair(it.second, it.first + it.second) }
+        .map { it.first }
+        .take(10)
+        .toList()
+
+    println(fibonacciList)
```

Comment:
It works 😄 But, you're using twice the memory you'd use if you just created the series without duplications.

Reply:
I fixed it so that it now works with yield.